rt_gene/src/rt_gene/gaze_estimation_models_pytorch.py

```diff
@@ -1,177 +1,150 @@
+        n_channels = [base_channels, base_channels * 2, base_channels * 4]
+
+        self.conv = nn.Conv2d(input_shape[1],
+                              n_channels[0],
+                              kernel_size=(3, 3),
+                              stride=1,
+                              padding=1,
+                              bias=False)
+
+        self.stage1 = self._make_stage(n_channels[0],
+                                       n_channels[0],
+                                       n_blocks_per_stage,
+                                       PreactResnet.BasicBlock,
+                                       stride=1)
+        self.stage2 = self._make_stage(n_channels[0],
+                                       n_channels[1],
+                                       n_blocks_per_stage,
+                                       PreactResnet.BasicBlock,
+                                       stride=2)
+        self.stage3 = self._make_stage(n_channels[1],
+                                       n_channels[2],
+                                       n_blocks_per_stage,
+                                       PreactResnet.BasicBlock,
+                                       stride=2)
+        self.bn = nn.BatchNorm2d(n_channels[2])
+
+        self._initialize_weights(self.modules())
 
     @staticmethod
-    def _create_fc_layers(num_out, num_features, fc=None):
-        if fc is None:
-            fc = [1024, 512, 256]
-
-        classifier = nn.Sequential(
-            nn.Linear(num_features, fc[0]),
-            nn.BatchNorm1d(fc[0]),
-            nn.ReLU(inplace=True),
-            nn.Linear(fc[0], fc[1]),
-            nn.BatchNorm1d(fc[1]),
-            nn.ReLU(inplace=True),
-            nn.Linear(fc[1], fc[2]),
-            nn.BatchNorm1d(fc[2]),
-            nn.ReLU(inplace=True),
-            nn.Linear(fc[2], num_out)
-        )
-        return classifier
+    def _initialize_weights(module):
+        if isinstance(module, nn.Conv2d):
+            nn.init.kaiming_normal_(module.weight, mode='fan_out')
+        elif isinstance(module, nn.BatchNorm2d):
+            nn.init.ones_(module.weight)
+            nn.init.zeros_(module.bias)
+        elif isinstance(module, nn.Linear):
+            nn.init.zeros_(module.bias)
 
     @staticmethod
-    def _init_weights(modules):
-        for m in modules:
-            if isinstance(m, nn.Linear):
-                nn.init.kaiming_uniform_(m.weight, mode="fan_in", nonlinearity="leaky_relu")
-                nn.init.zeros_(m.bias)
+    def _make_stage(in_channels, out_channels, n_blocks,
+                    block, stride):
+        stage = nn.Sequential()
+        for index in range(n_blocks):
+            block_name = "block{}".format(index + 1)
+            if index == 0:
+                stage.add_module(block_name, block(in_channels, out_channels, stride=stride))
+            else:
+                stage.add_module(block_name, block(out_channels, out_channels, stride=1))
```

Comment:
`stride=stride`?

Reply:
as in set the stride of the block to equal the function's "stride" parameter....sorry it's bad argument naming. Have a look at `PreactResnet` __init__ function; there are several calls to _make_stage(...stride=...)

Reply:
I'm not sure whether I fully understand, but as long as it's the intended use it's all good with me :)